### PR TITLE
DRILL-6020 Fix NullPointerException when querying JSON untyped path w…

### PR DESCRIPTION
**Pull request by MitchelLabonte**
…ith Union setting on

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/FieldIdUtil.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public static TypedFieldId getFieldIdIfMatches(ValueVector vector, TypedFieldId.
     } else if(v instanceof ListVector) {
       ListVector list = (ListVector) v;
       return getFieldIdIfMatches(list, builder, addToBreadCrumb, seg.getChild());
-    } else if (v instanceof  UnionVector) {
+    } else if (v instanceof  UnionVector && !seg.isLastPath()) {
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/FieldIdUtil.java]: Good catch! 
I think we should add check for nulls into method `getFieldIdIfMatchesUnion()` as it was done for `getFieldIdIfMatches()`. Also please add a unit test for this change. You may use [testFieldWithDots()](https://github.com/apache/drill/blob/acc5ed927e1fa4011ac1c3724d15197484b9f45b/exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java#L705) as an example.

But one more point: with this change error is not thrown, but only nulls are returned. I think we should also fix this issue on the borders of this pull request.

**Comment by MitchelLabonte** [comment]
> But one more point: with this change error is not thrown, but only nulls are returned. I think we should also fix this issue is the scope of this pull request.

@vvysotskyi I am not sure what to do to fix this. Any suggestion?

**Comment by vvysotskyi** [comment]
@MitchelLabonte, I think this NPE is just a consequence of the bug that should be fixed. Please investigate why Drill is trying to use child `PathSegment` when a value has VarChar type.

**Comment by MitchelLabonte** [comment]
@vvysotskyi
This is happening because the type is cached as a json object from the previous row. The fix is similar to the getFieldIdIfMatches() method so it looks like this is intended behaviour.
As you can see in the unit test, the results are what is expected after the fix. I am not sure what else could be done.

**Comment by vvysotskyi** [comment]
Sorry, I read your removed comment and made a suggestion that this test fails. 
Did you run all unit tests to see that this change does not break anything?

**Comment by MitchelLabonte** [comment]
@vvysotskyi No problem, yes all tests pass now.

**Comment by vvysotskyi** [comment]
@MitchelLabonte, thanks for the pull request, +1

**Pull request closed without merging**